### PR TITLE
Windows: Dynamically link Boost DLLs (avoid static linking) and ship them in sbin/

### LINKED ISSUE #7431: 2.11 RC1: Nessus Scan crash the Windows-Client.
## Describe the bug
We tested the 2.11 RC1 against the Nessus Scan again. So it's a litte bit a follow up from #6559: the Linux Client survived on CentOS 7 - Great! But not on our Windows System. Here the service is stopped after the scan. 2.11 RC1 is installed on a Windows Server 2012 R2 Standard. 

If you Need something more pls tell us.

## To Reproduce
Scan with a security scanner against a Windows system

## Expected behavior
The Windows Client should also not Crash like the Linux Client. 

## Your Environment
Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): 2.11 RC1 
* Operating System and version: Windows Server 2012 R2 Standard with .NET Framework 4.8

### PR DESCRIPTION
**Pull request by dnsmichi**
This comes with the motivation to avoid bundling Boost context
into our main binary. Whenever the context library allocates
stack memory, or does something weird, we want to see this
in the stack traces as origin from an external DLL and not the
main icinga2.exe binary.

TL;DR - this doesn't solve #7431 but makes debugging and development
a bit more easy. And it follows the Windows approach with DLLs.

### CODE REVIEW

**Comment by dnsmichi** [comment]
Thanks to @LordHepipud for the inspiration 👍

**Comment by dnsmichi** [comment]
Has no direct impact on the issue. 

The tests currently fail because they do not find the DLLs in the build installation directory in `debug\Bin\Debug\Debug`.

**Comment by dnsmichi** [comment]
Does not add to the solution, and I don't see any benefit in the future either.
